ee/desktop/runner/runner.go

```diff
@@ -212,40 +231,91 @@ func (r *DesktopUsersProcessesRunner) Interrupt(interruptError error) {
 	}
 }
 
+// Update handles control server updates for the desktop subsystem
+// This includes data related to the runner itself, and the user processes
 func (r *DesktopUsersProcessesRunner) Update(data io.Reader) error {
-	var desktopStatus client.DesktopUserStatus
+	var desktopStatus DesktopUserStatus
 	if err := json.NewDecoder(data).Decode(&desktopStatus); err != nil {
 		return fmt.Errorf("failed to decode desktop user control data: %w", err)
 	}
 
-	if desktopStatus.Status == "" {
-		return fmt.Errorf("empty desktop status")
+	// Regardless, we will write the menu data out to a file that can be grabbed by
+	// any desktop user processes, either when they refresh, or when they are spawned.
+	r.writeMenuFile(desktopStatus.Menu)
+
+	if r.processSpawningEnabled && !desktopStatus.Enabled {
+		// Desktop is currently enabled, control server is asking us to disable
+		level.Debug(r.logger).Log("msg", "runner is disabling desktop users processes")
+
+		// Stop automatically spawning desktop user processes
+		r.processSpawningEnabled = false
+
+		// Kill any existing desktop user processes, and we're done here
+		r.killDesktopProcesses()
+		return nil
+	}
+
+	if !r.processSpawningEnabled && desktopStatus.Enabled {
+		// Desktop is currently disabled, control server is asking us to enable
+		level.Debug(r.logger).Log("msg", "runner is enabling desktop users processes")
+
+		// Next execute loop tick will begin spawning desktop user processes
+		r.processSpawningEnabled = true
+		return nil
+	}
+
+	// If we're here:
+	// A) Desktop is currently disabled, and control server wants it to remain disabled
+	// B) Desktop is currently enabled, and control server wants it to remain enabled
+	//
+	// In case A), there will be no process records and thus no servers to talk to
```

Comment:
I think this might read more clearly if we return early in case A, rather than relying on these docs to make it clear why it's fine to proceed?

Reply:
I like that! I can shorten this up.